ocaml/typing/typecore.ml

```diff
@@ -4351,7 +4354,7 @@ and type_expect_
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_fun (l, Some default, spat, sbody) ->
-      assert(is_optional l); (* default allowed only with optional argument *)
+      assert(is_optional (Typetexp.transl_label l)); (* default allowed only with optional argument *)
```

Comment:
This is a bit ugly, calling `transl_label` just to call `is_optional`. There's unfortunately not a great place to put a function like `is_optional` that operates on `Parsetree`. Maybe just make a new function in `Btype` that operates on `Parsetree.arg_label`. But don't call `transl_label` here.